nav2_dynamic_params/test/test_dynamic_params_client.cpp

```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <memory>
+#include <sstream>
+#include <string>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "nav2_dynamic_params/dynamic_params_client.hpp"
+#include "rclcpp/rclcpp.hpp"
+
+class RclCppFixture
+{
+public:
+  RclCppFixture() {rclcpp::init(0, nullptr);}
+  ~RclCppFixture() {rclcpp::shutdown();}
+};
+
+RclCppFixture g_rclcppfixture;
+
+class DynamicParamsClientTest : public nav2_dynamic_params::DynamicParamsClient
+{
+public:
+  explicit DynamicParamsClientTest(rclcpp::Node::SharedPtr node)
+  : DynamicParamsClient(node)
+  {}
+
+  void call_test_event(std::string path, rclcpp::Parameter param, bool is_new = false)
+  {
+    auto event = std::make_shared<rcl_interfaces::msg::ParameterEvent>();
+    if (is_new) {
+      event->new_parameters.push_back(param.to_parameter_msg());
+    } else {
+      event->changed_parameters.push_back(param.to_parameter_msg());
+    }
+    event->node = path;
+    event_callback(event);
+  }
+};
+
+class ClientTest : public ::testing::Test
+{
+public:
+  ClientTest()
+  {
+    node_ = rclcpp::Node::make_shared("dynamic_param_client_test");
+    dynamic_params_client_ = std::make_unique<DynamicParamsClientTest>(node_);
+  }
+
+protected:
+  std::unique_ptr<DynamicParamsClientTest> dynamic_params_client_;
+  rclcpp::Node::SharedPtr node_;
+  bool callback_result_ = false;
+};
+
+
+TEST_F(ClientTest, testAddParameters)
+{
+  node_->set_parameters({rclcpp::Parameter("baz", 1)});
+
+  dynamic_params_client_->add_parameters();
+  dynamic_params_client_->add_parameters({"foobar"});
+
+  auto dynamic_param_map = dynamic_params_client_->get_param_map();
+  EXPECT_EQ(1, dynamic_param_map.count("/dynamic_param_client_test/baz"));
+  EXPECT_EQ(1, dynamic_param_map.count("/dynamic_param_client_test/foobar"));
```

Comment:
Should also check that it was not added to other namespaces besides `/dynamic_param_client_test`?

Reply:
At this point, the `dynamic_params_client` is completely unaware of other namespaces, so I don't know if that check is appropriate in this case